### PR TITLE
Adds Kotlin interop, Javacord implementation, Nexus, Dotenv and Docker

### LINKED ISSUE #14: Containerized the application
**Motivation**
- To ensure consistency across different machines (for local development)
- Prepare the bot for easy hosting

**Proposed Solution**

- [ ] Add Dockerfile
- [ ] Include docker build in the CI
    - alternatively, we can move this to a separate issue

### PR DESCRIPTION
**Pull request by ShindouMihou**
This PR adds Kotlin interop, basic implementation of Javacord and Nexus with a custom, native-Kotlin reflective Dotenv together with a simple multi-stage Dockerfile that should help upstart this project. It has been tested to a decent amount and should function properly.

Closes #14 and #10

### CODE REVIEW

**Comment by Gerv-G** [comment]
image

File: src/main/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<configuration>
```
Comment (Gerv-G) [inline src/main/resources/logback.xml:1]: do we have a YAML alternative for this?

Comment (ShindouMihou) [reply src/main/resources/logback.xml:1]: Nope, Logback is all XML AFAIK.

**Review by Gerv-G — APPROVED** [review]
LGTM despite not understanding some stuff 🤣 
Anyway, this is just the starting point and we can incrementally deliver or apply fixes where necessary.

**Pull request merged**